apps/ci/fsw/unit_test/ci_config_tbl_test.c

```diff
@@ -50,15 +50,124 @@
+    int32 expected = CFE_TBL_ERR_NO_ACCESS;
 
+    /* fail the register app */
+    Ut_CFE_TBL_SetReturnCode(UT_CFE_TBL_REGISTER_INDEX, expected, 1);
+
+    /* Execute the function being tested */
+    result = CI_InitTbls();
+
+    /* Verify results */
+    UtAssert_True (result == expected,
+                   "InitTbls(), fail TblRegister");
 }
 
 
-void CI_Config_Tbl_Test_AddTestCases(void)
+/**
+ * Test CI InitTbls(), fail TblLoad
+ */
+void Test_CI_InitTbls_Fail_TblLoad(void)
+{
+    /* Set a fail result */
+    int32 result = CFE_SUCCESS;
+    int32 expected = CFE_TBL_INFO_UPDATE_PENDING;
+
+    /* fail the register app */
+    Ut_CFE_TBL_SetReturnCode(UT_CFE_TBL_LOAD_INDEX, expected, 1);
+
+    /* Execute the function being tested */
+    result = CI_InitTbls();
+
+    /* Verify results */
+    UtAssert_True (result == expected,
+                   "InitTbls(), fail TblLoad");
+}
+
+
+/**
+ * Test CI InitTbls(), fail TblManage
+ */
+void Test_CI_InitTbls_Fail_TblManage(void)
+{
+    /* Set a fail result */
+    int32 result = CFE_SUCCESS;
+    int32 expected = CFE_TBL_ERR_INVALID_HANDLE;
+
+    /* fail the register app */
+    Ut_CFE_TBL_SetReturnCode(UT_CFE_TBL_MANAGE_INDEX, expected, 1);
+
+    /* Execute the function being tested */
+    result = CI_InitTbls();
+
+    /* Verify results */
+    UtAssert_True (result == expected,
+                   "InitTbls(), fail TblManage");
+}
+
+
+/**
+ * Test CI InitTbls(), fail TblGetAddress
+ */
+void Test_CI_InitTbls_Fail_TblGetAddress(void)
 {
-    UtTest_Add(CI_Config_Tbl_Test_Case1, CI_Test_Setup, CI_Test_TearDown, "CI_Config_Tbl_Test_Case1");
+    /* Set a fail result */
+    int32 result = CFE_SUCCESS;
+    int32 expected = CFE_TBL_ERR_NEVER_LOADED;
+
+    /* fail the register app */
+    Ut_CFE_TBL_SetReturnCode(UT_CFE_TBL_GETADDRESS_INDEX, expected, 1);
+
+    /* Execute the function being tested */
+    result = CI_InitTbls();
+
+    /* Verify results */
+    UtAssert_True (result == expected,
+                   "InitTbls(), fail TblGetAddress");
 }
 
 
+/**
+ * Test CI InitTbls(), Nominal
+ */
+void Test_CI_InitTbls_Nominal(void)
+{
+    /* Set a fail result */
+    int32 result = (CFE_SEVERITY_BITMASK & CFE_SEVERITY_ERROR)
+                   | CFE_EXECUTIVE_SERVICE | CFE_ES_ERR_APP_REGISTER;
+    int32 expected = CFE_SUCCESS;
+
+    /* Execute the function being tested */
+    result = CI_InitTbls();
+
+    /* Verify results */
+    UtAssert_True (result == expected, "InitTbls(), Nominal");
+}
+
+
+
+void CI_Config_Tbl_Test_AddTestCases(void)
+{
+    UtTest_Add(Test_CI_InitTbls_Fail_TblRegister, CI_Test_Setup, CI_Test_TearDown,
+               "Test_CI_InitTbls_Fail_TblRegister");
+    UtTest_Add(Test_CI_InitTbls_Fail_TblLoad, CI_Test_Setup, CI_Test_TearDown,
+               "Test_CI_InitTbls_Fail_TblLoad");
+    UtTest_Add(Test_CI_InitTbls_Fail_TblManage, CI_Test_Setup, CI_Test_TearDown,
+               "Test_CI_InitTbls_Fail_TblManage");
+#if 1  // Issue ticket #218
```

Comment:
Remove these macros.  Provided an issue ticket has been submitted, allow the test case to fail so whoever resolves the ticket does not have to remember to also remove these macros.  This is particularly important when integrating with more sophisticated Continuous Integration systems like Bamboo and Jira.  These CI tools can automatically flag an issue as resolved when it detects that the unit test goes from failed to passing.

Reply:
Yes sir.

Reply:
Done and pushed